packages/base-map/src/index.js

```diff
@@ -0,0 +1,248 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { LayersControl, Map, MapLayer, Popup, TileLayer } from "react-leaflet";
+import L from "leaflet";
+
+// eslint-disable-next-line func-names
+L.Evented.addInitHook(function() {
+  if (this) this.singleClickTimeout = null;
+  this.on("click", this.scheduleSingleClick, this);
+  this.on("dblclick dragstart zoomstart", this.cancelSingleClick, this);
+});
+
+L.Evented.include({
+  cancelSingleClick() {
+    // This timeout is key to workaround an issue where double-click events
+    // are fired in this order on some touch browsers: ['click', 'dblclick', 'click']
+    // instead of ['click', 'click', 'dblclick']
+    setTimeout(this.clearSingleClickTimeout.bind(this), 0);
+  },
+
+  scheduleSingleClick(e) {
+    this.clearSingleClickTimeout();
+
+    this.singleClickTimeout = setTimeout(
+      this.fireSingleClick.bind(this, e),
+      this.options.singleClickTimeout || 500
+    );
+  },
+
+  fireSingleClick(e) {
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.originalEvent._stopped) {
+      this.fire("singleclick", L.Util.extend(e, { type: "singleclick" }));
+    }
+  },
+
+  clearSingleClickTimeout() {
+    if (this.singleClickTimeout !== null) {
+      clearTimeout(this.singleClickTimeout);
+      this.singleClickTimeout = null;
+    }
+  }
+});
+
+class BaseMap extends Component {
+  componentDidMount() {
+    const lmap = this.refs.map.leafletElement;
+    lmap.options.singleClickTimeout = 250;
+    lmap.on("singleclick", this.onLeftClick);
+  }
+
+  componentDidUpdate() {}
+
+  // remove custom overlays on unmount
+  // TODO: Is this needed? It may have something to do with mobile vs desktop views
+  componentWillUnmount() {
+    const lmap = this.refs.map.leafletElement;
+    lmap.eachLayer(lmap.removeLayer);
```

Comment:
Hey Binh, 

I'm getting a "this is undefined" error here.  Stepping into Leaflet's eachLayer -> removeLayer functions, I can see that the 'this' in removeLayer is indeed null (see image).

![Screen Shot 2019-12-15 at 1 31 12 PM](https://user-images.githubusercontent.com/64323/70869382-3624bf00-1f3f-11ea-9618-9176973a8012.png)

BTW, yum install pulled in the following:
Leaflet 1.5.1 (2019-05-08)
React-Leaflet v2.6.0 (2019-11-18) (note that the changelog mentions "- Updated Leaflet dependency to v1.6.").

Take care,
Frank

Reply:
Hi Frank, I reverted an inlined function call (see my latest commit). Let me know if this error is still happening on your side.